File: StoneCarveManager.Model/Requests/OrderUpdateRequest.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace StoneCarveManager.Model.Requests
{
    public class OrderUpdateRequest
    {
        public int? AssignedEmployeeId { get; set; }

        public OrderStatus? Status { get; set; }

        public string? CustomerNotes { get; set; }

        public string? AdminNotes { get; set; }

        public string? AttachmentUrl { get; set; }

        public DateTime? EstimatedCompletionDate { get; set; }

        public DateTime? CompletedAt { get; set; }

        public string? DeliveryAddress { get; set; }

        public string? DeliveryCity { get; set; }

        public string? DeliveryZipCode { get; set; }

        public DateTime? DeliveryDate { get; set; }

        public List<OrderItemUpdateRequest>? Items { get; set; }
    }

     public enum OrderStatus
    {
        Pending,
        Processing,
        Shipped,
        Delivered,
        Cancelled,
        Returned
    }


}

File: UI/stonecarve_manager_desktop/lib/models/order_update_request.dart
class OrderUpdateRequest {
  int? assignedEmployeeId;
  int? status;
  String? customerNotes;
  String? adminNotes;
  String? attachmentUrl;
  DateTime? estimatedCompletionDate;
  DateTime? completedAt;
  String? deliveryAddress;
  String? deliveryCity;
  String? deliveryZipCode;
  DateTime? deliveryDate;
  List<OrderItemUpdateRequest>? items;

  OrderUpdateRequest({
    this.assignedEmployeeId,
    this.status,
    this.customerNotes,
    this.adminNotes,
    this.attachmentUrl,
    this.estimatedCompletionDate,
    this.completedAt,
    this.deliveryAddress,
    this.deliveryCity,
    this.deliveryZipCode,
    this.deliveryDate,
    this.items,
  });

  Map<String, dynamic> toJson() => {
    if (assignedEmployeeId != null) 'assignedEmployeeId': assignedEmployeeId,
    if (status != null) 'status': status,
    if (customerNotes != null) 'customerNotes': customerNotes,
    if (adminNotes != null) 'adminNotes': adminNotes,
    if (attachmentUrl != null) 'attachmentUrl': attachmentUrl,
    if (estimatedCompletionDate != null)
      'estimatedCompletionDate': estimatedCompletionDate?.toIso8601String(),
    if (completedAt != null) 'completedAt': completedAt?.toIso8601String(),
    if (deliveryAddress != null) 'deliveryAddress': deliveryAddress,
    if (deliveryCity != null) 'deliveryCity': deliveryCity,
    if (deliveryZipCode != null) 'deliveryZipCode': deliveryZipCode,
    if (deliveryDate != null) 'deliveryDate': deliveryDate?.toIso8601String(),
    if (items != null) 'items': items?.map((e) => e.toJson()).toList(),
  };
}

class OrderItemUpdateRequest {
  // Define fields as per your backend model
  // Example:
  int? id;
  int? stoneId;
  int? quantity;
  double? unitPrice;
  double? totalPrice;
  String? specifications;

  OrderItemUpdateRequest({
    this.id,
    this.stoneId,
    this.quantity,
    this.unitPrice,
    this.totalPrice,
    this.specifications,
  });

  Map<String, dynamic> toJson() => {
    if (id != null) 'id': id,
    if (stoneId != null) 'stoneId': stoneId,
    if (quantity != null) 'quantity': quantity,
    if (unitPrice != null) 'unitPrice': unitPrice,
    if (totalPrice != null) 'totalPrice': totalPrice,
    if (specifications != null) 'specifications': specifications,
  };
}
